### PR TITLE
#67 implements cookie banner

### LINKED ISSUE #67: Cookie banner for HCaptcha
HCaptcha uses cookies for that we need a Cookie banner.
When the cookie banner is displayed all input (tabbing, scrolling, clicking) should be blocked - until the cookies are accepted.

- [x] cookie banner
- [x] input is blocked
- [x] translation

@quintus-luis please provide a design.

### PR DESCRIPTION
**Pull request by 94726**
Closes #67 

This implements a banner prompting the user to consent the usage of cookies.
Wasn't sure which store to use for the cookie consent. It should probably be in a new general/miscellaneous store.

This is my first time doing React btw, please don't be too harsh :D.
Appreciate the feedback though.

Todos:
- [x] fix admin page buttons

### CODE REVIEW

File: src/models/index.ts [hunk 1]
```diff
@@ -14,3 +14,4 @@ export interface VoteOption {
 }
 
 export type Theme = "light" | "dark";
+export type CookieConsent = "accepted" | undefined
```
Comment (LeoTuet) [inline src/models/index.ts:17]: There was no new line - and github marked it
```suggestion
export type CookieConsent = "accepted" | undefined

```

File: src/components/CustomButton.module.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  color: var(--vote-button-color);
+  border-radius: 12px;
+  font: inherit;
+  font-weight: bold;
+  border: none;
+  outline: none;
+  padding: 1rem;
+  cursor: pointer;
+  transition: 0.2s ease-out;
+
```
Comment (LeoTuet) [inline src/components/CustomButton.module.scss:16]: An indication when the button is focused would be nice. The border should be black for light theme and black for the dark one.
```suggestion
  &:focus-visible {
    border-style: solid;
    border-color: var(--vote-button-focused)
  }
```

**Review by LeoTuet — CHANGES_REQUESTED** [review]
Is it on purpose that you can still scroll and tab through the app?

File: src/components/CustomButton.module.scss (hunk 2 above)
Comment (94726) [reply src/components/CustomButton.module.scss:16]: > An indication when the button is focused would be nice. The border should be black for light theme and black for the dark one.

There is an issue for that (#77) as I didn't want to make design assumptions and focus was purposefully disabled previously

File: src/models/index.ts (hunk 1 above)
Comment (94726) [reply src/models/index.ts:17]: lol

File: src/components/CustomButton.module.scss (hunk 2 above)
Comment (Aloso) [reply src/components/CustomButton.module.scss:16]: Since there's a TODO item and an issue for it, I think it's ok to defer this to a later PR.

**Review by LeoTuet — APPROVED** [review]
LGTM

File: src/components/BottomBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+export const BottomBanner = ({
+  text,
+  buttonText,
+  onButtonClick,
+}: BottomBannerProps) => {
+  const bannerRef = useRef<HTMLDivElement | null>(null);
+
+  const focusHandler = (event: KeyboardEvent) => {
```
Comment (quintus-luis) [inline src/components/BottomBanner.tsx:18]: im not quire sure why you would need this?

Comment (Aloso) [reply src/components/BottomBanner.tsx:18]: It's to to trap focus. A better solution would be to just use a `<dialog>`, but that's not supported everywhere yet. Or you could use a [library](https://github.com/focus-trap/focus-trap-react).

File: src/hooks/useCookieConsent.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+export const useCookieConsent = () => {
+  const dispatch = useDispatch();
+  const { cookieConsent } = useSelector(selectUI);
+
+  useEffect(() => {
+    const consentValue = window.localStorage.getItem("cookieConsent");
+
+    if (consentValue == "accepted")
+      dispatch(actions.setCookieConsent(consentValue));
```
Comment (Aloso) [inline src/hooks/useCookieConsent.ts:14]: Please don't omit the `{`braces`}`.

File: src/sections/LegalSection.module.scss [hunk 5]
```diff
@@ -1,16 +1,13 @@
...
-$large: 992px;
+@use "src/theme/mixin";
 
 .legalSubheader {
   margin: 0;
   font-size: 1.4rem;
 }
 
-@media (min-width: $large) {
+@media (min-width: 992px) {
```
Comment (Aloso) [inline src/sections/LegalSection.module.scss:8]: What's the reason for this change?

**Review by Aloso — CHANGES_REQUESTED** [review]
I think the HCaptcha shouldn't be shown until after cookies have been accepted, to make sure it can't create cookies before the user consented.

File: src/components/BottomBanner.tsx (hunk 3 above)
Comment (94726) [reply src/components/BottomBanner.tsx:18]: I implemented it as @LeoTuet begged me to do so and forgot dialog exists nowadays. Also don't think the browser support is that bad

File: src/sections/LegalSection.module.scss (hunk 5 above)
Comment (94726) [reply src/sections/LegalSection.module.scss:8]: Because I though it's weird to have breakpoint variables locally in one file, while only using them once.
It probably is best to define global breakpoints similarly to #56.

File: src/components/BottomBanner.tsx (hunk 3 above)
Comment (Aloso) [reply src/components/BottomBanner.tsx:18]: @94726 I would say browser support is still poor. It is supported in Firefox 98 and in Safari for macOS/iOS 15.4, but they're so new that almost nobody uses them yet. According to [caniuse](https://caniuse.com/dialog), only 62% of German users use a browser with `<dialog>` support.

File: src/sections/LegalSection.module.scss (hunk 5 above)
Comment (Aloso) [reply src/sections/LegalSection.module.scss:8]: That makes sense, thanks for explaining!

File: src/components/BottomBanner.tsx (hunk 3 above)
Comment (94726) [reply src/components/BottomBanner.tsx:18]: Oh you're totally, didn't know it's possible to look at the usage at caniuse
